com.unity.ml-agents/Runtime/SideChannels/StatsSideChannel.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+namespace MLAgents.SideChannels
+{
+    public class StatsSideChannel : SideChannel
```

Comment:
Would TensorBoardMetricsSideChannel be a more appropriate name?

Reply:
It won't just be for TB - with this change it'll show up anywhere Python stats are written (e.g. CSV, timers, and perhaps later the Console).

Reply:
Yeah, I kept the name generic but called out TensorBoard in the comments. I'll also add an example of this in the "Using TensorBoard" docs.

Reply:
Ok, let's then document the different ways that adding a _stat_ here will bubble up on the Python side. I find the term stats odd since TB refers to metrics, but I also can't see a better alternative (esp that it does align with timer stats)

Reply:
I moved the comments below up here and clarified a bit.